### PR TITLE
[16.0][IMP] sale_automatic_workflow: add send invoice option

### PR DESCRIPTION
**Pull request by johnny-longneck**
This PR adds another checkbox to workflows in order to send invoices automatically to customers.

This solves issue [2640](https://github.com/OCA/sale-workflow/issues/2640)

Added additional tests for job bypassing to get code coverage.
(Check if workflow process doesn't trigger the same records multiple times)

Setup:
- Install module
- Create a sale workflow
- Check "Send invoice"
- The default filter already checks for invoices that are not sent, you can change the filter to your needs
- Create sale order with this workflow
- Create invoice or let it be created from your workflow, validate it.
- The invoice will be sent during next workflow job execution

### CODE REVIEW

**Comment by johnny-longneck** [comment]
Can someone help me here - maybe @guewen ? 
The tests are failing because of no recpipient error.
I ran the same tests on my local setup and everything passed.

Furthermore I tried to test sale_automatic_workflow on runboat in several branches, and it did not seem to work.
What is going on? (Was expecting at least automatic invoices worked)

Thanks a lot for your help!

File: sale_automatic_workflow/models/automatic_workflow_job.py
Comment (johnny-longneck) [inline sale_automatic_workflow/models/automatic_workflow_job.py]: This was just cleanup from previous commit, while I was figuring out why tests failed.

**Review by rousseldenis — APPROVED** [review]
LGTM.

@johnny-longneck Could you also update DESCRIPTION.rst ? Thanks

**Comment by johnny-longneck** [comment]
> LGTM.
> 
> @johnny-longneck Could you also update DESCRIPTION.rst ? Thanks

Done, thanks for the hint and thanks for the review!

**Review by tmewes — APPROVED** [review]

**Comment by johnny-longneck** [comment]
@rousseldenis , sorry for disturbing you. Can you merge this PR when you find the time for it?
I would like to use it on a production installment.

Thank you!

**Comment by john-herholz-dt** [comment]
@grindtildeath are you maintainer for this?
I submitted this PR a long time ago and I would like to merge and migrate it.

For me I found it strange to have this cool sale_automatic_workflow module without sending customers their invoices.

File: sale_automatic_workflow/models/automatic_workflow_job.py
Comment (grindtildeath) [inline sale_automatic_workflow/models/automatic_workflow_job.py:137]: These values must already be set in `action_context`

File: sale_automatic_workflow/models/automatic_workflow_job.py
Comment (grindtildeath) [inline sale_automatic_workflow/models/automatic_workflow_job.py:144]: These values must also be set in the wizard through the `default_` keys in `action_context`

File: sale_automatic_workflow/models/automatic_workflow_job.py
Comment (grindtildeath) [inline sale_automatic_workflow/models/automatic_workflow_job.py]: AFAIU, you don't need to call these as long as you call `onchange_is_email` below.

File: sale_automatic_workflow/models/automatic_workflow_job.py
Comment (grindtildeath) [inline sale_automatic_workflow/models/automatic_workflow_job.py]: I'm not sure to understand the goal here, because AFAIU if condition is validated, the user must already be the superuser, or am I missing something?

**Review by grindtildeath — COMMENTED** [review]
Hello @john-herholz-dt 

I'm not the maintainer although I did work a lot on this module. But this is a nice contribution anyway.

Here are a few things to check though.

Also would be nice if you can add the overrides in [sale_automatic_workflow_job](https://github.com/OCA/sale-workflow/tree/16.0/sale_automatic_workflow_job) so that it works seamlessly with queue_job installed.

File: sale_automatic_workflow/models/automatic_workflow_job.py
Comment (john-herholz-dt) [reply sale_automatic_workflow/models/automatic_workflow_job.py:144]: I thought so, too but the tests fail if I remove it.

File: sale_automatic_workflow/models/automatic_workflow_job.py
Comment (john-herholz-dt) [reply sale_automatic_workflow/models/automatic_workflow_job.py]: I removed it, thanks!

File: sale_automatic_workflow/models/automatic_workflow_job.py
Comment (john-herholz-dt) [reply sale_automatic_workflow/models/automatic_workflow_job.py]: I removed this, too. Thanks!

**Comment by john-herholz-dt** [comment]
@rafaelbn , I saw you also merge PRs here.
My PR is open for a long time and got several reviews. The feature itself is already used by me in production.
Can you please check this PR when you have time and initialize the merging?

The sale_automatic_workflow module is pretty cool but it was lacking an essential feature: after generating invoices, also send them to the customers.

**Comment by rafaelbn** [comment]
@rousseldenis is PSC here and he validate this PR so better he should proceed. 😄 ❤️

**Comment by john-herholz-dt** [comment]
Can someone please finally merge this PR?

**Comment by john-herholz-dt** [comment]
@rousseldenis Do you think a review and PR is soon possible, otherwise I will close this PR and move it as a standalone module as addition to sale_automatic_workflow.
The Odoo instances I am involved are already on 17.0 and I need to move on.

**Comment by johnny-longneck** [comment]
Finally! Thank you so much @dreispt !

**Pull request merged**
